### DISCUSSION TITLE
Implementation of shell and beam elements

### DISCUSSION THREAD

**Discussion by alexanderkjellgren** [mfem/mfem#4553] [Q&A]
I would like to implement Mindlin-Reissner shell elements (2D elements in 3D space) and Timoshenko beam elements (1D elements in 3D space). 

Is this possible and where is a good point to start?
Would it be possible to connect beam and shell elements into a structure?

Thanks!

/AK

**Comment by tzanio** [comment]
There are several other groups interested in this. I will CC them below to get their input:

- @tupek2
- @mehranmfe

**Reply by mehranmfe** [reply]
We recently implemented the MITC shell element in our in-house library, which is built on MFEM. Our approach is an 'engineering' implementation rather than a 'mathematical' one, as we did not leverage MFEM's existing integrators to construct the elemental stiffness matrix. Currently, our shell element supports only linear triangular elements.

**Reply by linlujin0319-source** [reply]
> We recently implemented the MITC shell element in our in-house library, which is built on MFEM. Our approach is an 'engineering' implementation rather than a 'mathematical' one, as we did not leverage MFEM's existing integrators to construct the elemental stiffness matrix. Currently, our shell element supports only linear triangular elements.

Hi @mehranmfe, is there any example demonstrating the use of the MITC shell element?
